### PR TITLE
UKVI2 HOCS-1582 : correspondence cases are sent to correct workstack

### PR DESCRIPTION
**Pull request by MarkWrightBjss**
New endpoint for BPMN to change Team based on "Text" link type.
Merge to UKVI2.

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClientTest.java [hunk 1]
```diff
@@ -80,4 +86,19 @@ public void recreateStage(){
         verifyNoMoreInteractions(restHelper);
 
     }
+
+    @Test
+    public void updateTeamByStageAndTexts(){
+        String[] texts = { "Text1", "Text2" };
+        String expectedUrl = String.format("/case/%s/stage/%s/teamTexts", caseUUID, stageUUID);
+        UpdateCaseworkTeamStageTextRequest expectedBody = new UpdateCaseworkTeamStageTextRequest(caseUUID, stageUUID, stageType, "teamUUIDKey", "teamNameKey", texts);
```
Comment (hubert-raich) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClientTest.java:94]: expectedBody is not used?

File: src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClient.java [hunk 2]
```diff
@@ -112,6 +112,14 @@ public UUID updateStageTeam(UUID caseUUID, UUID stageUUID, UUID teamUUID, String
         return teamUUID;
     }
 
+    public Map<String, String> updateTeamByStageAndTexts(UUID caseUUID, UUID stageUUID, String stageType, String teamUUIDKey, String teamNameKey, String[] texts) {
+        UpdateCaseworkTeamStageTextRequest request = new UpdateCaseworkTeamStageTextRequest(caseUUID, stageUUID, stageType, teamUUIDKey, teamNameKey, texts);
+        String test = String.format("/case/%s/stage/%s/teamTexts", caseUUID, stageUUID);
```
Comment (hubert-raich) [inline src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClient.java:117]: ?, We can remove it:)

File: src/test/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClientTest.java (hunk 1 above)
Comment (MarkWrightBjss) [reply src/test/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClientTest.java]: Good spot. Removed.

File: src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClient.java (hunk 2 above)
Comment (MarkWrightBjss) [reply src/main/java/uk/gov/digital/ho/hocs/workflow/client/caseworkclient/CaseworkClient.java]: LOL.

**Pull request merged**